src/dyn/dyn_variables_electronic.cpp

```diff
@@ -825,28 +825,62 @@ void dyn_variables::init_active_states(bp::dict _params, Random& rnd){
   ///================= Actual calculations  ================================
 
   act_states.clear();  
+  act_states_dia.clear();  
 
-  //# Dynamical variables
-  //================ For adiabatic - set up them directly =================
-  for(traj=0; traj<ntraj; traj++){
-    if(init_type==0 || init_type==1 || init_type==4){ 
-      act_states.push_back(istate);
-    }
-    else if(init_type==2 || init_type==3){
+  if(rep==1){

```

Comment:
this is the `init_active_states` function - the idea of it was that it would ONLY init what is expected - only adiabatic or only diabatic properties, NOT update the other representation. the `update_` functions are meant to be called after it to update other reps. This would keep the logic of the functions simple - not implying more than it should be doing. Please remove the updates from this function and add them to the corresponding `update_` functions. Incorporate that in the workflow you use.